hawkbit-sdk/hawkbit-sdk-device/src/main/java/org/eclipse/hawkbit/sdk/device/DdiTenant.java

```diff
@@ -40,17 +41,26 @@ public void destroy() {
         controllers.clear();
     }
 
-    public DdiController create(final Controller controller, final UpdateHandler updateHandler) {
+    public DdiController createController(final Controller controller, final UpdateHandler updateHandler) {
         final DdiController ddiController = new DdiController(tenant, controller, updateHandler, hawkbitClient);
         controllers.put(controller.getControllerId(), ddiController);
         return ddiController;
     }
 
-    public void remove(final String controllerId) {
-        Optional.ofNullable(controllers.remove(controllerId)).ifPresent(DdiController::stop);
+    public void deleteController(final String controllerId) {
+        Optional.ofNullable(controllers.remove(controllerId)).ifPresent(removedController -> hawkbitClient.mgmtService(MgmtTargetRestApi.class, tenant).deleteTarget(removedController.getControllerId()));
```

Comment:
Do we need indeed to mix Mgmt API calls here ? 
Ddi Controller is supposed to connect, poll and handle updates eventually. Not sure if we need to add deletion behaviour here. Of course I'm open for discussions about it.

Reply:
yeah.. maybe we should discuss the scope of DDITenant/DMFTenant and what to have as sdk 
i personally would like to have such functionalities within the sdk of devices (even though it does call some management apis.. but in the end it is related to the devices)

It also makes sense to separate it.. but as SDK to have only option to register without removing is somewhat half sdk to me..